File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <bits/stdc++.h>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    int cnt[10] = {0,};
    while(n != 0){
        ++cnt[n % 10];
        n /= 10;
    }
    
    for (int i = 9; i >= 0; --i){
        while(cnt[i] != 0){
            answer = answer * 10 + i;
            --cnt[i];
        }
    }
    return answer;
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
public class Solution {
    public static long solution(long n) {
        long answer = 0;
        int[] cnt = new int[10];
        
        while (n != 0) {
            cnt[(int)(n % 10)]++;
            n /= 10;
        }

        for (int i = 9; i >= 0; i--) {
            while (cnt[i] != 0) {
                answer = answer * 10 + i;
                cnt[i]--;
            }
        }
        return answer;
    }
}